app/src/main/java/org/asteroidos/sync/services/SynchronizationService.java

```diff
@@ -31,198 +33,241 @@
+        editor.putString(MainActivity.PREFS_DEFAULT_MAC_ADDR, device.getAddress());
+        mDevice = device;
+        String name = mDevice.getName();
+        try {
+            Message answer = Message.obtain(null, MSG_SET_LOCAL_NAME);
+            answer.obj = name;
+            replyTo.send(answer);
+            replyTo.send(Message.obtain(null, MSG_SET_STATUS, mState, 0));
+        } catch (RemoteException | NullPointerException ignored) {
+        }
+        editor.putString(MainActivity.PREFS_DEFAULT_LOC_NAME, name);
+        editor.apply();
     }
 
-    void handleReqBattery() {
-        if(mDevice == null) return;
-        if(mState == STATUS_DISCONNECTED) return;
-        mDevice.read(AsteroidUUIDS.BATTERY_UUID, new BleDevice.ReadWriteListener()
-        {
-            @Override public void onEvent(ReadWriteEvent result)
-            {
-                if(result.wasSuccess())
-                    try {
-                        replyTo.send(Message.obtain(null, MSG_SET_BATTERY_PERCENTAGE, result.data()[0], 0));
-                    } catch (RemoteException | NullPointerException ignored) {}
+    final void handleUpdate() {
+        if (mDevice != null) {
+            try {
+                replyTo.send(Message.obtain(null, MSG_SET_STATUS, mState, 0));
+            } catch (RemoteException | NullPointerException ignored) {
             }
-        });
+        }
     }
 
-    void handleSetDevice(String macAddress) {
-        SharedPreferences.Editor editor = mPrefs.edit();
-        editor.putString(MainActivity.PREFS_DEFAULT_MAC_ADDR, macAddress);
-
-        if(macAddress.isEmpty()) {
-            if(mState != STATUS_DISCONNECTED) {
-                mScreenshotService.unsync();
-                mWeatherService.unsync();
-                mNotificationService.unsync();
-                mMediaService.unsync();
-                mTimeService.unsync();
-                mDevice.disconnect();
-                mDevice.unbond();
-            }
-            mDevice = null;
-            editor.putString(MainActivity.PREFS_DEFAULT_LOC_NAME, "");
-        } else {
-            mDevice = mBleMngr.getDevice(macAddress);
+    final public void unsyncServices() {
+        bleServices.forEach(IService::unsync);
+        nonBleServices.forEach(IService::unsync);
+    }
 
-            String name = mDevice.getName_normalized();
-            try {
-                Message answer = Message.obtain(null, MSG_SET_LOCAL_NAME);
-                answer.obj = name;
-                replyTo.send(answer);
+    final public void syncServices() {
+        bleServices.forEach(IService::sync);
+        nonBleServices.forEach(IService::sync);
+    }
 
-                replyTo.send(Message.obtain(null, MSG_SET_STATUS, mState, 0));
-            } catch (RemoteException | NullPointerException ignored) {}
+    @Override
+    public final ConnectionState getConnectionState() {
+        ConnectionState state = null;
+        switch (mState) {
```

Comment:
Mhh am I missing something here ? ConnectionState seems to be an enum with only three possible values: connected, connecting, disconnected, here you have a long switch for these three cases but you only assign state the same values.

Wouldn't a simple "return mState" work the same ?

Reply:
yup, we had two different state implementations (I think one is from the old implementation and I didn't want to break other stuff). I replaced everything now to use the new `ConnectionState` enum.